packages/paste-core/components/progress-bar/src/ProgressBar.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import {Box, type BoxProps} from '@twilio-paste/box';
+import {useProgressBar} from '@twilio-paste/react-spectrum-library';
+import {animated, useSpring} from '@twilio-paste/animation-library';
+import {keyframes} from '@twilio-paste/styling-library';
+
+const AnimatedBox = animated(Box);
+const IndeterminateKeyframes = keyframes`
+  from {
+    left: -10%;
+  }
+  to {
+    left: 105%;
+  }
+`;
+
+export interface ProgressBarProps {
```

Comment:
issue: these need to extend the base HTML element types. Whether thats `HTMLPasteProps<'progressbar'>` or something else will be down to the overlap of Spectrums custom element vs the native.